android/library/src/main/java/io/logto/android/authflow/browser/BrowserAuthFlow.kt

```diff
@@ -0,0 +1,114 @@
+package io.logto.android.authflow.browser
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import io.logto.android.api.LogtoService
+import io.logto.android.callback.AuthenticationCallback
+import io.logto.android.config.LogtoConfig
+import io.logto.android.constant.AuthConstant
+import io.logto.android.pkce.Util
+import io.logto.android.utils.UrlUtil
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+
+object BrowserAuthFlow {
+
+    private var flowProperty: BrowserAuthFlowProperty? = null
+
+    fun init(
+        logtoConfig: LogtoConfig,
+        authenticationCallback: AuthenticationCallback
+    ): BrowserAuthFlow {
+        resetFlow()
+
+        val codeVerifier = Util.generateCodeVerifier()
+        val authUrl = generateAuthUrl(logtoConfig, codeVerifier)
+
+        flowProperty = BrowserAuthFlowProperty(
+            codeVerifier,
+            authUrl,
+            logtoConfig,
+            authenticationCallback
+        )
+
+        return this
+    }
+
+    fun startAuth(context: Context) {
+        startAuthActivity(context)
+    }
+
+    fun onBrowserResult(redirectUri: Uri?) {
+        flowProperty?.let { property ->
+            if (redirectUri == null) {
+                property.authenticationCallback
+                    .onFailed(Error("onBrowserResult missing redirectUri!"))
```

Comment:
please add a linear issue for more precise error like https://github.com/logto-io/js/pull/30#discussion_r717813381

Reply:
sure. TBD in [LOG-80](https://linear.app/silverhand/issue/LOG-80/implement-authentication-exceptions-according-to-logto-core)